### PR TITLE
TokenFactory bindings #NTRN-431

### PR DESCRIPTION
**Pull request by foxpy**
Foundation for:
- https://github.com/neutron-org/neutron-dev-contracts/pull/16
- https://github.com/neutron-org/neutron-integration-tests/pull/147

Expected merge flow:
1. we review and merge this particular PR
2. we release neutron-sdk v0.5.0
3. I target PR in neutron-dev-contracts to use neutron-sdk v0.5.0
4. we review and merge other two PRs, which both depend on this PR (which will have already been merged in step 1, at this point)

[[Jira task](https://p2pvalidator.atlassian.net/browse/NTRN-431)] [[Recent test run](https://github.com/neutron-org/neutron-tests/actions/runs/4831402575)]

### CODE REVIEW

File: packages/neutron-sdk/src/bindings/msg.rs [hunk 1]
```diff
@@ -126,6 +126,36 @@ pub enum NeutronMsg {
...
+    },
+    /// TokenFactory message.
+    /// Contracts can burn native tokens for an existing factory denom
+    /// that they are the admin of.
+    /// Currently, the burn from address must be the admin contract.
+    BurnTokens {
+        denom: String,
+        amount: Uint128,
+        /// Must be set to `""` for now
+        burn_from_address: String,
```
Comment (foxpy) [inline packages/neutron-sdk/src/bindings/msg.rs:156]: @albertandrejev why is it so? I have copy pasted this part from your code in neutron wasmbindings.

Comment (albertandrejev) [reply packages/neutron-sdk/src/bindings/msg.rs:156]: It was copy pasted from osmosis wasmbinding as well. https://github.com/osmosis-labs/osmosis/blob/e871beddfd4b7abc473c6fd7b9f5cfafbe9769b8/wasmbinding/bindings/msg.go#L46

File: packages/neutron-sdk/src/bindings/msg.rs [hunk 2]
```diff
@@ -14,8 +14,8 @@ use serde_json_wasm::to_string;
 /// IbcFee defines struct for fees that refund the relayer for `SudoMsg` messages submission.
 /// Unused fee kind will be returned back to message sender.
 /// Please refer to these links for more information:
-/// IBC transaction structure - https://docs.neutron.org/neutron/interchain-txs/messages/#msgsubmittx
-/// General mechanics of fee payments - https://docs.neutron.org/neutron/feerefunder/overview/#general-mechanics
+/// IBC transaction structure - <https://docs.neutron.org/neutron/interchain-txs/messages/#msgsubmittx]>
```
Comment (albertandrejev) [inline packages/neutron-sdk/src/bindings/msg.rs:17]: extra `]` at the end of this line

**Review by quasisamurai — APPROVED** [review]
nice

**Review by oopcode — APPROVED** [review]

**Pull request merged**